WebApp/Controllers/HomeController.cs

```diff
@@ -141,6 +142,27 @@ public async Task<ActionResult> ReadMail()
             }
         }
 
+        /// <summary>
+        /// Challenge the user based on the exception.
+        /// </summary>
+        /// <param name="exc"></param>
+        private void ChallengeUser(MicrosoftIdentityWebChallengeUserException exc)
```

Comment:
@jennyf19 would you think it would be a good addition to ID.Web/ OWIN?
Maybe with a dictionary as a parameter if people want to add more properties?